### PR TITLE
#608 - ProxyRepoConfig and YamlProxyRepoConfig

### PR DESCRIPTION
**Pull request by olegmoz**
Part of #608 
Added `ProxyRepoConfig` interface and `YamlProxyRepoConfig` implementation

### CODE REVIEW

File: src/main/java/com/artipie/YamlProxyRepoConfig.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Proxy repository config from YAML.
+ *
+ * @since 0.12
+ */
+public final class YamlProxyRepoConfig implements ProxyRepoConfig {
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/YamlProxyRepoConfig.java:40]: @olegmoz I'd suggest to shorten the name of this class to simple `YamlProxy`, we can understand that this is repo config from the interface name.

File: src/test/java/com/artipie/YamlProxyRepoConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for {@link YamlProxyRepoConfig}.
+ *
+ * @since 0.12
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public final class YamlProxyRepoConfigTest {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/YamlProxyRepoConfigTest.java:42]: @olegmoz I'd suggest to add test for absent `remotes` section, as far as I see `orElseGet` branch from `YamlProxyRepoConfig` line 61 is not tested...

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@olegmoz two suggestions, take a look please

File: src/main/java/com/artipie/YamlProxyRepoConfig.java (hunk 1 above)
Comment (olegmoz) [reply src/main/java/com/artipie/YamlProxyRepoConfig.java]: @olenagerasimova thanks, the name is actually too long. But I feel that `YamlProxy` would be misguiding: this object is not a proxy, but a config. So I changed it to `YamlProxyConfig` and moved to `.repo` package

**Review by olenagerasimova — APPROVED** [review]
@olegmoz thanks, looks good to merge

**Comment by 0crat** [comment]
Job `gh:artipie/artipie#685` is not assigned, can't get performer

**Pull request merged**